portfolio/src/main/webapp/assets/js/testimonials.js

```diff
@@ -110,4 +110,104 @@ function renderVoiceButton(){
 function onPageLoad(){
     getTestimonials();
     initializeVoiceControl();
-}
\ No newline at end of file
+}
+
+/** Takes two words and builds/returns the starting DP table to work with */
+function buildInitialTable(w1, w2){
+    // create 2d matrix dp table using array of arrays
+    dp = [];
+    
+    // create first row representing edit distance of empty vs w2
+    r1 = []
+    for(i=0; i<w2.length+1; i++){
+      r1.push(i);
+    }
+    
+    // add first row to dp table
+    dp.push(r1);
+
+    // Fill in remaining rows TODO: optimize to O(N) rather than O(N^2)
+    for(i=1; i < w1.length+1; i++){
+        row = [i];
+        for(j=0; j<w2.length; j++){
+          row.push(null);
+        }
+        dp.push(row);
+    }
+    return dp;
+}
+
+function findEditDistance(w1,w2){
+  dp = buildInitialTable(w1,w2);
+
+  for(j=1; j<dp.length; j++){
+    for(k=1; k<dp[j].length; k++){
+
+      // if character in w1 matches character in w2 give current cell same value as diagnol
+      if(w1[j-1] == w2[k-1]){
+        dp[j][k]=dp[j-1][k-1];
+        
+      // otherwise give current cell value = 1 + minimum (diagnol, top, left)
+      } else {
+        diagnol = dp[j-1][k-1];
+        above = dp[j-1][k];
+        left = dp[j][k-1];
+        dp[j][k] = 1 + Math.min(diagnol, above, left);
+      }
+    }
+  }
+
+  // return bottom right corner as the edit distance btwn w1 and w2
+  return(dp[dp.length-1][dp[0].length-1]);
+}
+
+
+/** Takes array of strings and a term. Returns an array of objects 
+ *  (TODO: sorted by their edit distance to the term. Where first item in the array has the lowest edit distance.) 
+ *  maxED specifies the string with maximum edit distance to the term to be included in the returned array.
+ *  Note that for this purpose, edit distance refers to the minimum edit distance of a word in the string
+ */
+function distanceOfArr(arrSentences, term, maxED){
+  let retArray = []
+  for (const sentence of arrSentences){
+    ed = minEditDistanceWord(sentence, term)
+    if(ed < maxED){
+      retArray.push({"sentence": sentence, "editDistance": ed});
+    }
+
+  }
+  
+  return retArray;
+}
+
+/** Takes a sentence and returns the minimum edit distance of a word in the sentence to the term */
+function minEditDistanceWord(sentence, term){
+  arrOfWords = sentence.split(" ");
+
+  let minEditDistance = Infinity;
+  for (const word of arrOfWords){
+    minEditDistance = Math.min(minEditDistance, findEditDistance(word, term));
+  }
+
+  return minEditDistance
+}
+
+
+/**
+Since I haven't implemented yet into actual site, here is a test that I ran:
```

Comment:
Nice! I would include additional tests here.

Eventually we should test in a separate file, but this is fine for now.